--- a/qtbase/src/plugins/platforms/eglfs/deviceintegration/eglfs_omapfb/qeglfsomapfbintegration.cpp
+++ b/qtbase/src/plugins/platforms/eglfs/deviceintegration/eglfs_omapfb/qeglfsomapfbintegration.cpp
@@ -1,10 +1,12 @@
 #include "qeglfsomapfbintegration.h"
 #include <QDebug>
+#include <linux/fb.h>
 #include <linux/omapfb.h>
 #include <sys/ioctl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <unistd.h>
 #include <private/qcore_unix_p.h>
 #include <cerrno>
 #include <cstring>
@@ -228,4 +230,28 @@
   }
 }
 
+void *QEglFSOmapfbIntegration::nativeResourceForIntegration(const QByteArray &resource)
+{
+  QByteArray lowerCaseResource = resource.toLower();
+
+  // This "interface" is stolen from
+  // github.com/mer-hybris/qt5-qpa-hwcomposer-plugin/blob/master/hwcomposer/qeglfsintegration.cpp
+
+  if (lowerCaseResource == "displayoff") {
+    blankDisplay(true);
+  } else if (lowerCaseResource == "displayon") {
+    // Called from lipstick to turn on the display (src/homeapplication.cpp)
+    blankDisplay(false);
+  }
+
+  return 0;
+}
+
+void QEglFSOmapfbIntegration::blankDisplay(bool blank)
+{
+  int arg = blank ? FB_BLANK_POWERDOWN : FB_BLANK_UNBLANK;
+
+  ioctl(m_fd, FBIOBLANK, arg);
+}
+
 QT_END_NAMESPACE
--- a/qtbase/src/plugins/platforms/eglfs/deviceintegration/eglfs_omapfb/qeglfsomapfbintegration.h
+++ b/qtbase/src/plugins/platforms/eglfs/deviceintegration/eglfs_omapfb/qeglfsomapfbintegration.h
@@ -27,8 +27,11 @@
   qreal refreshRate() const Q_DECL_OVERRIDE;
   Qt::ScreenOrientation nativeOrientation() const Q_DECL_OVERRIDE;
   bool hasCapability(QPlatformIntegration::Capability cap) const Q_DECL_OVERRIDE;
+  void *nativeResourceForIntegration(const QByteArray &resource) Q_DECL_OVERRIDE;
 
 private:
+  void blankDisplay(bool blank);
+
   int m_fd;
   bool m_manual;
   QSize m_screenSize;
